std/algorithm/sorting.d

```diff
@@ -387,8 +387,8 @@ See_Also:
 */
 Range partition(alias predicate,
         SwapStrategy ss = SwapStrategy.unstable, Range)(Range r)
-    if ((ss == SwapStrategy.stable && isRandomAccessRange!(Range))
-            || (ss != SwapStrategy.stable && isForwardRange!(Range)))
+    if ((ss == SwapStrategy.stable && isRandomAccessRange!(Range) && hasLength!Range && hasSlicing!Range)
+            || isInputRange!(Range))
```

Comment:
for a  _stable_ swap strategy RandomAccess is needed, we shouldn't relax this concern ;-)